In this challenge the task was to insert 10 exceptions in the assembply code generated by the AAPG.
In the .yaml file, there is an option to insert different types of exceptions under exception generation section.
To isert 10 exception I made ecause00-ecause10 as 1, so that an exception of each kind will be generated in the test.
![ecause-1](https://github.com/vyomasystems-lab/riscv-ctb-challenge-SudeepJoshi22/blob/main/challenge_level2/challenge2_exceptions/challenge2_2_gen_10_exceptions)
Afteer the inserting exceptions, there should be proper trap handlers to handle those exceptions.
So custom_trap_handler is made true.
![customtrap](https://github.com/vyomasystems-lab/riscv-ctb-challenge-SudeepJoshi22/blob/main/challenge_level2/challenge2_exceptions/challenge2_2_custom_trap_true.png)

If we don't make custom trap handler as true, the exceptions will lead to infinite loop and won't generate dump file.
![customtrapfalse](https://github.com/vyomasystems-lab/riscv-ctb-challenge-SudeepJoshi22/blob/main/challenge_level2/challenge2_exceptions/challenge2_2_custop_trap_false.png)

We an see illegal instructions inserted below.
![ecause9](https://github.com/vyomasystems-lab/riscv-ctb-challenge-SudeepJoshi22/blob/main/challenge_level2/challenge2_exceptions/challenge2_2%20ecause1.png)
![ecause4](https://github.com/vyomasystems-lab/riscv-ctb-challenge-SudeepJoshi22/blob/main/challenge_level2/challenge2_exceptions/challenge2_2_ecause2.png)
